dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/supplier/FileResourceSupplier.java

```diff
@@ -87,4 +99,15 @@ private void collectResourceIds( List<String> fileResourceAttributes, List<Strin
             }
         } );
     }
+
+    private void collectResourceIds( List<String> fileResourceDataElements, List<String> fileResourceIds,
+        Set<DataValue> dataElements )
+    {
+        dataElements.forEach( de -> {
+            if ( fileResourceDataElements.contains( de.getDataElement() ) && !StringUtils.isEmpty( de.getValue() ) )
+            {
+                fileResourceIds.add( de.getValue() );
+            }
```

Comment:
if we add these filter, we need to test it all  cases like empty value and mixed cases with existing resource

Reply:
Done